### DISCUSSION TITLE
How to include lumino src into other proyect?

### DISCUSSION THREAD

**Discussion by Yairama** [LuminoEngine/Lumino#174] [Q&A] [answered]
Hello and thanks for open the git hub discussions, I have been testing lumino in a personal project but there are many things that I would like to add / change (like a new controller for the camera to the unity style, more input keys, customizable general components, etc) but from the project that built lumino (with the command lumino new "project") the src are not found to be able to modify them, is there any way to generate a project with the src included using the lumino command?
or some way to include lumino src in other projects?

**Comment by lriki** [comment, marked as answer]
Thanks for the comment!

Your project C++?

If so, I think you should build the Lumino library on your end and link it to your project.

For example, if you are developing in Visual Studio 2019, run the following commands in the repository root, in order

```
dotnet run -- BuildExternalProjects MSVC2019-x64-MT
dotnet run -- BuildEngine_MSVC MSVC2019-x64-MT
dotnet run -- MakeNativePackage
```

Then, the library files will be generated in the following folder.

- build/NativePackage/Engine/MSVC2019-x64-MT/lib

Third-party libraries should link to these. (zlib, bullet, etc.)

Libraries are to be linked by #pragma. After setting the library path, define LN_MSVC_AUTO_LINK_LIBRARIES.
https://github.com/LuminoEngine/Lumino/blob/master/include/LuminoEngine.hpp#L148-L184

The include path can be the include folder in the repository root.


The following file is the solution file. You can open it to modify and build the source code.

- build/MSVC2019-x64-MT/Lumino.sln

To use the built library right away, link the following libraries.

- build/MSVC2019-x64-MT/src/LuminoCore/Debug/LuminoCored.lib
- build/MSVC2019-x64-MT/src/LuminoCore/Release/LuminoCore.lib
- build/MSVC2019-x64-MT/src/LuminoEngine/Debug/LuminoEngine-staticd.lib
- build/MSVC2019-x64-MT/src/LuminoEngine/Release/LuminoEngine-static.lib

For other information on building, the following documents may be helpful.

https://github.com/LuminoEngine/Lumino/blob/master/docs/HowToBuild.md
https://github.com/LuminoEngine/Lumino/blob/master/.github/workflows/package-continuous.yml
